tests/integration/test_records.py

```diff
@@ -56,29 +58,37 @@ def test_float_vector(client):
     ensemble_id = _create_ensemble(client)
 
     records = [
-        ("name", "[1, 2, 3, 4, 5]", True),
-        ("name", "[5, 4, 3, 2, 1]", False),
-        ("fail", "yes", False),
+        ("name", "[1, 2, 3, 4, 5]", True, status.HTTP_200_OK),
+        ("name", "[5, 4, 3, 2, 1]", False, status.HTTP_409_CONFLICT),
+        ("fail", "yes", False, status.HTTP_415_UNSUPPORTED_MEDIA_TYPE),
     ]
 
     # Post each of the record
-    for name, data, should_succeed in records:
+    for name, data, should_succeed, status_code in records:
         failed = False
         try:
             resp = client.post(
                 f"/ensembles/{ensemble_id}/records/{name}",
                 data=data,
                 headers={"RecordType": "float_vector"},
             )
-            failed = resp.status_code != 200
+            failed = resp.status_code != status.HTTP_200_OK
         except Exception as exc:
             failed = True
         if should_succeed and failed:
```

Comment:
I realise I was the one who wrote this code, but on second look I think we should change `should_succeed` to `should_fail`, or change `failed` to `succeeded`, because otherwise it's confusing. Could you do this in this PR?